# /usr/bin/python3
# encode:-*- utf-8 -*-
# try:
#     print('try...')
#     r = 10 / 0
#     print('result:', r)
# except ZeroDivisionError as e:
#     print('except:', e)
# finally:
#     print('finally...')
# print('END')
# try:
#     a = input("请输入一个整数：")
#     r = 10 / int(a)
#     print('result:', r)
#
# except ValueError as e:         # 当输入非整数时,匹配异常
#     print('ValueError:', e)
# except ZeroDivisionError as e:  # 当输入为0时
#     print('ZeroDivisionError:', e)
# else:
#     print('no error!')
# finally:
#     print('finally...')
# print('END')
#
#
# class FooError(ValueError):
#     pass
#
#
# def foo(s):
#     n = int(s)
#     if n == 0:
#         raise FooError('invalid value: %s' % s)
#     return 10 / n
#
#
# try:
#     foo('0')
# except FooError as e:
#     print('FooError!')

#  def open(file, mode='r', buffering=None, encoding=None, errors=None, newline=None, closefd=True):
# Why does mode and no mode have the same effect
with open('ErrorData/1301001.dat', 'r+', encoding='gb2312') as dataFile:
    data = dataFile.read()
    print(data)
    dataFile.write("test for write...............")
    print(data)


